### PR TITLE
Support ad-hoc unit testing against existing Couchbase Server bucket

### PR DESCRIPTION
**Pull request by adamcfraser**
When running unit tests at dev time, there are scenarios where it's more efficient to test against an already provisioned bucket, instead of depending on the TestBucketPool to provision/flush the bucket.  To simplify this type of testing, this adds SG_TEST_USE_EXISTING_BUCKET environment variable which will bypass the bucket readier code, and instead attempt to establish a connection to the named bucket.

### CODE REVIEW

**Comment by torcolvin** [comment]
I like this idea, but I might wait for https://github.com/couchbase/sync_gateway/pull/5832 to land first.

**Comment by bbrks** [comment]
@torcolvin I'd rather get this in and deal with the conflict on #5832 as I've not yet fully completed the test bucket pool changes there

File: base/main_test_bucket_pool.go [hunk 1]
```diff
@@ -318,6 +322,44 @@ func (tbp *TestBucketPool) GetWalrusTestBucket(t testing.TB, url string) (b Buck
 	}
 }
 
+// GetExistingBucket opens a bucket conection to an existing bucket
+func (tbp *TestBucketPool) GetExistingBucket(t testing.TB) (b Bucket, s BucketSpec, teardown func()) {
+	testCtx := TestCtx(t)
+
+	bucketCluster := initV2Cluster(UnitTestUrl())
+
+	var useNamedCollections bool
```
Comment (torcolvin) [inline base/main_test_bucket_pool.go]: Is this `shouldUseNamedCollections` ?

File: base/main_test_bucket_pool.go [hunk 2]
```diff
@@ -318,6 +322,44 @@ func (tbp *TestBucketPool) GetWalrusTestBucket(t testing.TB, url string) (b Buck
...
+		if !TestsDisableGSI() {
+			useNamedCollections = true
+		}
+	}
+
+	bucketName := tbpBucketName(TestUseExistingBucketName())
+	bucketSpec := getBucketSpec(bucketName, useNamedCollections)
+
+	bucket := bucketCluster.Bucket(bucketSpec.BucketName)
+	err := bucket.WaitUntilReady(10*time.Second, nil)
```
Comment (torcolvin) [inline base/main_test_bucket_pool.go]: Can we turn this into a test constant https://cs.github.com/couchbase/sync_gateway/blob/d1ce9ca698e6e3a4474e74886fb41a5bd4648e6a/base/main_test_cluster.go?q=waituntilready#L149 ? I notice different times on server 6.6 and I'd rather modify in one spot.

**Review by torcolvin — COMMENTED** [review]
LGTM, just some comments about code duplication.

File: base/main_test_bucket_pool.go (hunk 2 above)
Comment (bbrks) [reply base/main_test_bucket_pool.go]: I think I'll just set everything to use the `waitForReadyBucketTimeout = time.Minute` const - will take slightly longer in the failure case but happy-path shouldn't be affected by the increased timeout.

**Review by torcolvin — APPROVED** [review]

**Pull request merged**